novaagent/libs/centos.py

```diff
@@ -192,12 +192,28 @@ def _restart_network_manager(self):
         :rtype: Popen
         :return: Popen of 'nmcli connection on'
         """
-        # We really don't care if we can't bring it down
-        try:
-            p = Popen(['nmcli', 'networking', 'off'])
-            p.wait()
-        except Exception:
-            pass
+        p = Popen(
+            ['nmcli', 'connection', 'reload'],
+            stdout=PIPE,
+            stderr=PIPE,
+            stdin=PIPE
+        )
+        p.wait()
+        if p.returncode != 0:
+            log.error(
+                "Unable to reload network configurations from ifcfg "
+                "files with 'nmcli connection reload'"
+            )
+            return p
+
+        p = Popen(['nmcli', 'networking', 'off'])
+        p.wait()
+        if p.returncode != 0:
+            log.error(
+                "Unable to bring down networking with "
+                "'nmcli networking off'"
+            )
+            return p
```

Comment:
Since you are going to return the process handle when the process failed, you should probably go ahead and pass `stdout=PIPE`, `stderr=PIPE`, and `stdin=PIPE` here. Really, just so that the calling code (which will get the process handle) can call `communicate()` on it and get back the bytes for the error so that it can get logged (on line 329).

Reply:
Agreed.  Updated and pushed.